pregen.js

```diff
@@ -91,3 +96,146 @@ export function preparePythonEnv(filePath, options) {
     }
   }
 }
+
+/**
+ * Method to check and prepare the environment for node
+ *
+ * @param {String} filePath Path
+ * @param {Object} options CLI Options
+ */
+export function prepareNodeEnv(filePath, options) {
+  // check tool for windows
+  for (const pt of options.projectType) {
+    const nodeVersion = pt.replace(/\D/g, "");
+    if (
+      pt.startsWith("nodejs") &&
+      nodeVersion &&
+      !process.env.NODE_INSTALL_ARGS
+    ) {
+      if (!isNvmAvailable()) {
+        if (process.env.NVM_DIR) {
+          // for scenarios where nvm is not present, but
+          // we have $NVM_DIR
+          // custom logic to find nvmNodePath
+          let nvmNodePath;
+          const possibleNodeDir = join(process.env.NVM_DIR, "versions", "node");
+
+          if (!tryLoadNvmAndInstallTool(nodeVersion)) {
+            console.log(
+              `Could not install Nodejs${nodeVersion}. There is a problem with loading nvm from ${process.env.NVM_DIR}`,
+            );
+            return;
+          }
+
+          const nodeVersionArray = readdirSync(possibleNodeDir, {
+            withFileTypes: true,
+          });
+          const nodeRe = new RegExp(`^v${nodeVersion}.`);
+          for (const nodeVersionsIter of nodeVersionArray) {
+            const fullPath = join(possibleNodeDir, nodeVersionsIter.name);
+            if (
+              nodeVersionsIter.isDirectory() &&
+              nodeRe.test(nodeVersionsIter.name)
+            ) {
+              nvmNodePath = join(fullPath, "bin");
+            }
+          }
+          if (nvmNodePath) {
+            doNpmInstall(filePath, nvmNodePath);
+          } else {
+            console.log(
+              `"node version ${nodeVersion} was not found. Please install it with 'nvm install ${nodeVersion}"`,
+            );
+            return;
+          }
+        } else {
+          console.log(
+            "Install nvm by following the instructions at https://github.com/nvm-sh/nvm",
+          );
+          return;
+        }
+      }
+      // set path instead of nvm use
+      const nvmNodePath = getNvmToolPath(nodeVersion);
+      doNpmInstall(filePath, nvmNodePath);
+    }
+  }
+}
+
+/**
+ * If NVM_DIR is in path, however nvm command is not loaded.
+ * it is possible that required nodeVersion is not installed.
+ * This function loads nvm and install the nodeVersion
+ *
+ * @param {String} nodeVersion required version number
+ *
+ * @returns {Boolean} true if successful, otherwise false
+ */
+export function tryLoadNvmAndInstallTool(nodeVersion) {
+  const NVM_DIR = process.env.NVM_DIR;
+
+  const command = `
```

Comment:
It's ok for this PR, but this won't work in Windows powershell. Maybe this could be added in another PR.